crates/services/p2p/src/codecs.rs

```diff
@@ -1,19 +1,24 @@
+pub mod bounded;
 pub mod postcard;
+pub mod unbounded;
 
-use crate::{
-    gossipsub::messages::{
-        GossipTopicTag,
-        GossipsubBroadcastRequest,
-        GossipsubMessage,
-    },
-    request_response::messages::{
-        RequestMessage,
-        V2ResponseMessage,
-    },
-};
+use crate::gossipsub::messages::GossipTopicTag;
 use libp2p::request_response;
+use serde::{
+    Deserialize,
+    Serialize,
+};
 use std::io;
 
+trait DataFormatCodec {
+    type Error;
+    fn deserialize<'a, R: Deserialize<'a>>(
+        encoded_data: &'a [u8],
+    ) -> Result<R, Self::Error>;
+
+    fn serialize<D: Serialize>(data: &D) -> Result<Vec<u8>, Self::Error>;
+}
```

Comment:
Are we sure there are no data formats that take parameters? I'd err on the cautious side and accept a `self` parameter to these methods. This would for example allow a dynamically configurable formalt such as:

```
enum MyFormat {
    Postcard,
    Json
}

While I don't know by heart when or where we'd like to use this, it feels overly restrictive to explicitly rule it out. This also eliminates the need for `PhantomData` markers in our structs, which is a bit of code smell to me.
```

```suggestion
trait DataFormatCodec {
    type Error;
    fn deserialize<'a, R: Deserialize<'a>>(
        &self,
        encoded_data: &'a [u8],
    ) -> Result<R, Self::Error>;

    fn serialize<D: Serialize>(&self, data: &D) -> Result<Vec<u8>, Self::Error>;
}
```

Reply:
Thanks, I like the suggestion. I have made the change here: c564941d769d1524d04dfec9675b6f7da7d5b5f6. Note that the `DataFormatCodec` trait has been removed in favour of two `Encode` and `Decode` traits, following what has been done in the `storage` crate.